vumi/components/message_store_migrators.py

```diff
@@ -53,17 +53,21 @@ def migrate_from_1(self, mdata):
         mdata.set_value('batches', mdata.old_data.get('batches', []))
         mdata.copy_values('message')
         mdata.copy_indexes('message_bin')
+        mdata.copy_indexes('message_with_status_bin')
 
         return mdata
 
     def reverse_from_2(self, mdata):
-        # We copy the `batches` field even though the older model version
-        # doesn't know about it. This lets us migrate v2 -> v1 -> v2 without
-        # losing data.
+        # We copy the `batches` field and related indexs even though the older
+        # model version doesn't know about them. This lets us migrate
+        # v2 -> v1 -> v2 without losing data.
```

Comment:
I would actually expect the indexes to go away if we switched back to `v1` -- i.e. I'd expect data to be lost by the round trip. Not sure that's enough of a reason to actually through the indexes away. I assume that wanting current tests to pass is what prompted this change initially?